base/cvd/cuttlefish/host/libs/web/caching_build_api.cpp

```diff
@@ -76,36 +76,37 @@ CachingPaths ConstructCachePaths(const std::string& cache_base,
 }  // namespace
 
 CachingBuildApi::CachingBuildApi(
-    std::unique_ptr<HttpClient> http_client,
-    std::unique_ptr<HttpClient> inner_http_client,
-    std::unique_ptr<CredentialSource> credential_source, std::string api_key,
-    const std::chrono::seconds retry_period, std::string api_base_url, std::string project_id,
-    const std::string cache_base_path)
-    : BuildApi(std::move(http_client), std::move(inner_http_client),
-               std::move(credential_source), std::move(api_key), retry_period,
-               std::move(api_base_url), std::move(project_id)),
-      cache_base_path_(std::move(cache_base_path)) {};
+    std::unique_ptr<IBuildApi> build_api, const std::string cache_base_path)
+              : build_api_(std::move(build_api)),
+                cache_base_path_(std::move(cache_base_path)) {};
 
 Result<bool> CachingBuildApi::CanCache(const std::string& target_directory) {
   return CF_EXPECT(CanHardLink(target_directory, cache_base_path_));
 }
 
+Result<Build> CachingBuildApi::GetBuild(const BuildString& build_string,
+                         const std::string& fallback_target) {
+  CF_EXPECT(build_api_ != nullptr);
```

Comment:
Put this in a `CHECK` in the constructor to avoid repeating it in every method call, something like:
```
CHECK(build_api_ != nullptr) << "Underlying build api can't be null";
```